wgsl/index.bs

```diff
@@ -493,22 +538,32 @@ sampler_comparison
 ## Texture Functions ## {#texture-functions}
 
 <pre class='def'>
-vec4<type> texture_load(texture_ro, f32 | vec(2|3)<f32> coords, i32 level_of_detail)
+`vec4<type> texture_load(texture_ro, i32 | vec(2|3)<i32> coords, i32 level_of_detail)`
   %32 = OpImageFetch %v4float %31 %29 Lod %int_2
 
-TODO(dsinclair): Add `texture_write` method for texture_wo
+TODO(dsinclair): Texture load on a sampled texture
+
+TODO(dsinclair): Texture load on multi-sampled texture
+
+TODO(dsinclair): Add `texture_write` method for texture_wo with integral coords
 
-vec4<type> texture_sample(texture_sampled, sampler, f32 | vec(2|3)<f32> coords)
+TODO(dsinclair): Allow a small constant offset on the coordinate? May not be portable.
+
+`vec4<type> texture_sample(texture_sampled, sampler, f32 | vec(2|3)<f32> coords)`
   %24 = OpImageSampleImplicitLod %v4float %19 %23
 
-vec4<type> texture_sample_level(texture_sampled, sampler, f32 | vec(2|3)<f32> coords, f32 lod)
+`vec4<type> texture_sample_level(texture_sampled, sampler, f32 | vec(2|3)<f32> coords, f32 lod)`
   %25 = OpImageSampleExplicitLod %v4float %19 %23 Lod %float_0
 
-vec4<type> texture_sample_bias(texture_sampled, sampler,  f32 | vec(2|3)<f32> coords, f32 bias)
+`vec4<type> texture_sample_bias(texture_sampled, sampler,  f32 | vec(2|3)<f32> coords, f32 bias)`
   %19 = OpImageSampleImplicitLod %v4float %14 %17 Bias %float_2
 
-f32 texture_sample_compare(texture_depth, sampler_comparison, f32 | vec(2|3)<f32> coords, i32 lod)
-  %65 = OpImageSampleDrefImplicitLod %float %62 %63 %64
+`f32 texture_sample_compare(texture_depth, sampler_comparison, f32 | vec(2|3)<f32> coords, i32 lod)`
```

Comment:
looks like my earlier comment got lost somehow.
We need to have an argument here for the "depth reference".
Also, we can't have `i32 lod` here, since Metal doesn't like that (only 0.0 is supported anyway).

Reply:
Done